standard/attributes.md

```diff
@@ -858,6 +860,275 @@ For invocations that occur within field or event initializers, the member name u
+>
+> <!-- Example: {template:"standalone-lib", name:"AllowNullAttribute", replaceEllipsis:true, customEllipsisReplacements:["\"XYZ\""]} -->
+> ```csharp
+> #nullable enable
+> public class X
+> {
+>     [AllowNull]
+>     public string ScreenName
+>     {
+>         get => _screenName;
+>         set => _screenName = value ?? GenerateRandomScreenName();
+>     }
+>     private string _screenName = GenerateRandomScreenName();
+>     private static string GenerateRandomScreenName() => ...;
+> }
+> ```
+>
+> Given the following use of that property’s set accessor
+>
+> ```csharp
+> var v = new X();
+> v.ScreenName = null;   // may warn without attribute AllowNull
+> ```
+>
+> without the attribute, the compiler may generate a warning because the non-nullable-typed property appears to be set to a null value. The presence of the attribute suppresses that warning. *end example*
+
+#### §The-DisallowNull-Attribute The DisallowNull attribute
+
+Specifies that a null value is disallowed as an input even if the corresponding type allows it.
+
+> *Example*: Consider the following property in which null is the default value, but clients can only set it to a non-null value.
+>
+> <!-- Example: {template:"standalone-lib", name:"DisallowNullAttribute"} -->
+> ```csharp
+> #nullable enable
+> public class X
+> {
+>     [DisallowNull]
+>     public string? ReviewComment
+>     {
+>         get => _comment;
+>         set => _comment = value ?? throw new ArgumentNullException(nameof(value),
+>            "Cannot set to null");
+>     }
+>     private string? _comment = default;
+> }
+> ```
+>
+> The get accessor could return the default value of `null`, so the compiler may warn that it must be checked before access. Furthermore, it warns callers that, even though it could be null, callers shouldn't explicitly set it to null. *end example*
+
+#### §The-DoesNotReturn-Attribute The DoesNotReturn attribute
+
+Specifies that a given method never returns.
+
+> *Example*: Consider the following:
+>
+> <!-- Example: {template:"standalone-lib", name:"DoesNotReturnAttribute"} -->
+> ```csharp
+> public class X
+> {
+>     [DoesNotReturn]
+>     private void FailFast() =>
+>         throw new InvalidOperationException();
+>
+>     public void SetState(object? containedField)
+>     {
+>         if ((!isInitialized) || (containedField == null))
+>         {
+>             FailFast();
+>         }
+>         // null check not needed.
+>         _field = containedField;
+>     }
+> 
+>     private bool isInitialized = false;
+>     private object _field;
+> }
+> ```
+>
+> The presence of the attribute helps the compiler in a number of ways. First, the compiler can issue a warning if there's a path where the method can exit without throwing an exception. Second, the compiler can consider any code after a call to that method as unreachable, until an appropriate catch clause is found. Third, the unreachable code won't affect any null states. *end example*
```

Comment:
This should be reworded. The compiler doesn't use this attribute for definite assignment nor reachability. However, it will only supress any nullability.